server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/materials/index.tsx

```diff
@@ -0,0 +1,107 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MithrilViewComponent} from "jsx/mithril-component";
+import _ from "lodash";
+import m from "mithril";
+import {Material, Materials} from "models/new_pipeline_configs/materials";
+import {Secondary} from "views/components/buttons";
+import {CollapsiblePanel} from "views/components/collapsible_panel";
+import {IconGroup} from "views/components/icons";
+import {Delete} from "views/components/icons";
+import {MaterialOperations} from "../../create_pipeline_page";
+import styles from "../index.scss";
+import materialStyles from "./index.scss";
+import {AddMaterialModal} from "./modals";
+
+interface Attrs {
+  materials: Materials;
+  materialOperations: MaterialOperations;
+}
+
+export class MaterialsWidget extends MithrilViewComponent<Attrs> {
+  view(vnode: m.Vnode<Attrs>) {
+    let materials = [];
+    let expanded  = true;
+    if (_.isEmpty(vnode.attrs) || _.isEmpty(vnode.attrs.materials) || vnode.attrs.materials.length === 0) {
+      expanded  = true;
+      materials = ["No materials defined"];
+    } else {
+      materials = [<table className={materialStyles.table} data-test-id={"materials-index-table"}>
+        <thead data-test-id="table-header">
+        <tr data-test-id="table-header-row">
+          <th>Material Name</th>
+          <th>Type</th>
+          <th>Url</th>
+          <th></th>
+        </tr>
+        </thead>
+        {this.materialsData(vnode)}
+      </table>];
+    }
+
+    return <CollapsiblePanel header="Materials"
+                             actions={
+                               <Secondary onclick={(e: MouseEvent) => this.addMaterialModal(e, vnode)}
+                                          data-test-id={"add-material-button"}>Add
+                                 Material</Secondary>
+                             }
+                             expanded={expanded}
+                             dataTestId="pipeline-materials-container">
+      {materials}
+    </CollapsiblePanel>;
+  }
+
+  materialsData(vnode: m.Vnode<Attrs>): any {
+    const tbodyContent = Array<any>();
+    vnode.attrs.materials.forEach((material) => {
+      const materialContent = Array<any>();
+      materialContent.push(<td>{material.name()}</td>);
+      materialContent.push(<td>{this.getTypeForDisplay(material.type())}</td>);
+      materialContent.push(<td>{material.materialUrl()}</td>);
+      materialContent.push(<td class={styles.textAlignRight}>
+        <IconGroup>
+          <Delete onclick={() => vnode.attrs.materialOperations.onDelete(material)} title={"delete-material"}
+                  data-test-id={"delete-material-button"}/>
+        </IconGroup>
+      </td>);
+      tbodyContent.push(<tr>{materialContent}</tr>);
+    });
+    return <tbody>{tbodyContent}</tbody>;
+  }
+
+  getTypeForDisplay(type: string | undefined) {
```

Comment:
Since this is repeated in [MaterialEditor](https://github.com/gocd/gocd/pull/6816/files#diff-358d82298e09369e31e9f98320b48e21R38), I think it can move to the model itself.

Reply:
I have re-used a constant `SUPPORTED_MATERIAL` which was already there in MaterialEditor.